msadmin/static/js/qa/main.js

```diff
@@ -6,10 +6,45 @@ function checkSnapshotValidity () {
 
 $('#saveButton').on('click', function() {
     var status = $('#my-pulldown3').val();
-    if (status === 'ready' && $(this).data('snapshot') == 0) {
-        $('#warningDialog').modal('show');
-    }
-    else {
-        $('#editProblem').submit();
+    var warningString = "";
+    var formElement = document.getElementById('editProblem');
+    if (formElement.checkValidity()) {
+        if (status === 'ready') {
+            if ($(this).data('snapshot') == 0) {
+                warningString += "snapshot";
+            }
+            if ($(this).data('hint') == "True") {
+                if (warningString.length) {
```

Comment:
Works for now, but in a future version, we can consider adding all problematic fields to an array and calling a join using ' or ' as a delimiter.